src/app/parcels/add/databaseFunctions.ts

```diff
@@ -0,0 +1,21 @@
+import supabase, { InsertSchema, Schema } from "@/supabase";
+
+export type ParcelDatabaseInsertRecord = InsertSchema["parcels"];
+type ParcelsDatabaseFetchRecord = Pick<Schema["parcels"], "primary_key" | "client_id">;
+
+export const insertParcel = async (
+    parcelRecord: ParcelDatabaseInsertRecord
+): Promise<ParcelsDatabaseFetchRecord> => {
+    const {
+        data: ids,
+        status,
+        error,
+    } = await supabase.from("parcels").insert(parcelRecord).select("primary_key, client_id");
+
+    if (error === null && Math.floor(status / 100) === 2) {
+        return ids![0];
+    }
+    throw Error(
```

Comment:
Just want to know whats the difference between throw Error and throw new Error and which one is better for this case?

Reply:
So they are very similar, there is some difference in that `throw new Error` creates an error object, and in `throw Error` the Error acts as a function that creates the error object. Apparently `throw new Error` is more technically correct, so will use that here!